### PR TITLE
Fixes #5186 - qualify ForEach() and Where() syntax

### LINKED ISSUE #5145: Documentation relating to upgrading PowerShell
There is zero information on upgrading a PowerShell install. It would be useful to include this in the Installation guidance pages

For Windows,
- Should you simply download and run a new MSI?
- What it you are using the newer MSIX packages?
- Can this be done from the CLI?
- What about different version for v6 and v7? 
- Other method of install, do they require different upgrade procedures?

For other OS:
- Re-install the package?
- Use a package manager?

etc....

---
#### Document details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 46356aae-d82e-63ce-63cf-fe10cd47d76d
* Version Independent ID: c5ccf493-dc82-bb65-830a-d5857048ce26
* Content: [Installing PowerShell Core on Windows - PowerShell](https://docs.microsoft.com/en-gb/powershell/scripting/install/installing-powershell-core-on-windows?view=powershell-6#feedback)
* Content Source: [reference/docs-conceptual/install/Installing-PowerShell-Core-on-Windows.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/docs-conceptual/install/Installing-PowerShell-Core-on-Windows.md)
* Product: **powershell**
* Technology: **powershell-conceptual**
* GitHub Login: @joeyaiello
* Microsoft Alias: **jaiello**

### LINKED ISSUE #5186: Parentheses are only optional with the script-block .ForEach() and .Where() overloads if the script block is the *only* argument
The linked topic currently states:

> The syntax requires the usage of a script block. Parentheses are optional.

If the script block is the _only_ argument, parentheses can indeed be omitted.

As an aside: Doing so is inadvisable, because it invites confusion with the `ForEach-Object` and `Where-Object` cmdlets and more generally between method (expression) syntax and argument (command) syntax and may therefore lead to the false expectation that a _space_ is allowed between between the method name and the script block.

```powershell
(1, 2).ForEach{1+$_}  # OK - script block is the only argument.

# But: There must be no space between the method name and the script block.
(1, 2).ForEach {1+$_}  # BOOM!
```

Parentheses can _not_ be omitted if you pass additional arguments - this should be clearly stated, especially given that neglecting to use parentheses doesn't cause a syntax error, but simply does something different:

```powershell
(1, 2, 1).Where{ 1 -eq $_ }   # OK - script block is the only argument.

# To pass additional arguments, the parentheses are REQUIRED
(1, 2, 1).Where({ 1 -eq $_ }, 'First')   

# NO BUENO: executes .Where(), then constructs 
# an array with 'First' as the last element.
(1, 2, 1).Where{ 1 -eq $_ },'First'
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e532b436-8796-e889-e3d9-baae466ba418
* Version Independent ID: ab878bfa-19e1-2011-aaf8-37456f420ddb
* Content: [about_Arrays - PowerShell](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_arrays?view=powershell-7#foreach)
* Content Source: [reference/7/Microsoft.PowerShell.Core/About/about_Arrays.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/7/Microsoft.PowerShell.Core/About/about_Arrays.md)
* Product: **powershell**
* Technology: **powershell-cmdlets**
* GitHub Login: @SteveL-MSFT
* Microsoft Alias: **slee**

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Fixes #5186  - qualify ForEach() and Where() syntax
Fixes [AB#1653248](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1653248)

## PR Context

Select the type(s) of documents being changed.

**Cmdlet reference & about_ topics**
- [x] Version 7 content
- [x] Version 6 content
- [x] Version 5.1 content

**Conceptual articles**
- [ ] Fundamental conceptual articles
- [ ] Script sample articles
- [ ] DSC articles
- [ ] Gallery articles
- [ ] JEA articles
- [ ] WMF articles
- [ ] SDK articles

## PR Checklist

- [x] I have read the [contributors guide](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/CONTRIBUTING.md) and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] PR is targeted at the _staging_ branch
- [x] All relevant versions updated
- [x] Includes content related to issues and PRs - see [Closing issues using keywords](https://help.github.com/en/articles/closing-issues-using-keywords).
- [x] This PR is ready to merge and is not **Work in Progress**
  - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the
    title and remove the prefix when the PR is ready.

### CODE REVIEW

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Arrays.md [hunk 1]
```diff
@@ -26,47 +26,47 @@ To create and initialize an array, assign multiple values to a variable. The
...
 
 ```powershell
 $A = 22,5,10,8,12,9,80
 ```
 
-You can also create and initialize an array by using the range operator (..).
-For example, to create and initialize an array named "$B" that contains the
-values 5 through 8, type:
+You can also create and initialize an array by using the range operator (`..`).
+The following example creates an array named containing the values 5 through 8.
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Arrays.md]: An array named what?

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Arrays.md [hunk 2]
```diff
@@ -26,47 +26,47 @@ To create and initialize an array, assign multiple values to a variable. The
...
-As a result, $B contains four values: 5, 6, 7, and 8.
+As a result, `$B` contains four values: 5, 6, 7, and 8.
 
-When no data type is specified, PowerShell creates each array as an
-object array (type: System.Object[]). To determine the data type of an array,
-use the GetType() method. For example, to determine the data type of the $a
-array, type:
+When no data type is specified, PowerShell creates each array as an object
+array (**System.Object[]**). To determine the data type of an array, use the
+GetType() method. For example, to determine the data type of the `$a` array,
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Arrays.md]: Method should be bold or fenced. Also, if this is the same array as farther above, do you want to use the same case?

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Arrays.md [hunk 3]
```diff
@@ -359,19 +365,19 @@ $a = @(
...
 ```
 
-```output
+```Output
 $a rank: 1
 ```
 
 Building a truly multidimensional array, in PowerShell, requires the
-assistance of the *.Net Framework*. Like in the following example:
+assistance of the .Net Framework. Like in the following example:
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Arrays.md]: That last sentence needs some work.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Arrays.md (hunk 1 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Arrays.md]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Arrays.md (hunk 2 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Arrays.md]: Fixed both

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Arrays.md (hunk 3 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Arrays.md]: Fixed

**Pull request merged**